packages/swapper/src/safe-utils/index.ts

```diff
@@ -10,10 +11,26 @@ import type {
 } from './types'
 
 export const fetchSafeTransactionInfo = async ({
-  chainId,
+  accountId,
   maybeSafeTxHash,
+  fetchIsSmartContractAddressQuery,
 }: FetchSafeTransactionArgs): Promise<SafeTxInfo> => {
+  if (!accountId) {
+    return { transaction: null, isSafeTxHash: false }
+  }
+  const chainId = fromAccountId(accountId).chainId
   if (!isEvmChainId(chainId)) return { transaction: null, isSafeTxHash: false }
+  const isSmartContractAddress = await fetchIsSmartContractAddressQuery(
+    fromAccountId(accountId).account,
+    chainId,
+  )
+
+  if (!isSmartContractAddress) {
+    // Assume any smart contract address is a SAFE, and by extension, any non-smart contract address is not
+    // This is super naive, but this is exactly the same way SAFE does it
+    // https://github.com/safe-global/safe-core-sdk/blob/ea0d5018a93f294dfd891e6c8963edcb96431876/packages/protocol-kit/src/Safe.ts#L303
+    return { transaction: null, isSafeTxHash: false }
+  }
```

Comment:
Also the guts of this PR, avoids spamming SAFE API, https://github.com/shapeshift/web/pull/7525/files#r1711302800